src/main/java/de/thetaphi/forbiddenapis/Checker.java

```diff
@@ -309,6 +309,11 @@ public ClassSignature lookupRelatedClass(String internalName, String internalNam
       }
     } catch (IOException ioe) {
       throw new RelatedClassLoadingException(ioe, Type.getObjectType(internalNameOrig).getClassName());
+    } catch (RuntimeException re) {
+      if (AsmUtils.isExceptionInAsmClassReader(re)) {
+        throw new RelatedClassLoadingException(re, Type.getObjectType(internalNameOrig).getClassName());
+      }
+      throw re;
```

Comment:
What if NPE happens somewhere in forbiddenapis?
Do you think it would be useful to have the relevant class name in the exception message?

Reply:
This issue should not deal with problems in forbiddenapis itsself. If there's a bug causing NPE, then its a whole different issue that should not be masked by this.

In that case theres no way to workaround and the issue needs fixing. This is only about buggy class files.

Reply:
I do not seek for workaround here. What I want is I want all failures from forbidden-apis to include the pointer to the class that causes the failure. In other words "NPE while analyzing abc.class" is way better than "NPE: null".

For instance, see below a crash I got from checkerframework. It does clarify which file is bad, so I know what should I change. If the only information I get from forbidden-api is "NPE in forbidden-api", that is not very helpful, especially, if the bug is caused by Javac failing to produce proper bytecode.

```
; The Checker Framework crashed.  Please report the crash.
  Compilation unit: /home/runner/work/calcite/calcite/core/src/main/java/org/apache/calcite/runtime/ResultSetEnumerable.java
  Last visited tree at line 110 column 15:
                return list.toArray();
  Exception: java.lang.Throwable; java.lang.Throwable
  	at org.checkerframework.javacutil.BugInCF.<init>(BugInCF.java:14)
  	at org.checkerframework.framework.type.visitor.AbstractAtmComboVisitor.defaultAction(AbstractAtmComboVisitor.java:52)
  	at org.checkerframework.framework.type.visitor.AbstractAtmComboVisitor.visitArray_Typevar(AbstractAtmComboVisitor.java:105)
  	at org.checkerframework.framework.util.AtmCombo.accept(AtmCombo.java:298)
```

Reply:
The lastVisitedTree logging could be added later, as this would indeed help.

But that needs a completely different setup (refactoring the checker to keep track of line numbers while parsing). In that case we could print some log error like checker framework.

About the current issue: I want to clearly separate issues caused by forbiddenapis and ASM parsing. Here it's a bug in the class file, not in forbiddenapis code.

I'll put this on the list of TODOs.

Reply:
Actually this code will catch also NullPointerException inside ASM ClassReader.

Reply:
Ah by the way: Here it's the wrong place to add anything like this at all: Your comment should be part of the other try...catch a few lines below. This one is only about loading related class signatures (references to some external class not actually checked). NPEs can't happen here :-)

Reply:
Would you like to add some `@Nullable` annotations? :-)

Of course, having line numbers (or bytecode instruction index) would be a big plus, however, having even file names would be great as well.

Reply:
I opened an issue about the additional "parsing state" printed to error logger in case of exception: #175